src/coreclr/jit/importercalls.cpp

```diff
@@ -4213,14 +4181,33 @@ GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic          intrinsic,
                         return gtNewDconNode(BitOperations::UInt32BitsToSingle(u32Cns), TYP_FLOAT);
                     }
                 }
-                else
+                // TODO-CQ: We should support this on 32-bit via decomposition
+                else if (TargetArchitecture::Is64Bit || (toType == TYP_FLOAT))
```

Comment:
can you explain to me the `toType == FLOAT` please

Reply:
On 32bit only 4 byte bitcast node in supported.